### PR TITLE
Add test to ensure edge cases around batch threadsize calculation

### PR DESCRIPTION
**Pull request by knutwalker**

### CODE REVIEW

File: tests/src/test/java/org/neo4j/graphalgo/core/utils/ParallelUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    @Test
+    public void threadSizeShouldReturn1ForZeroElements() throws Exception {
+        int batchSize = between(2, 1_000_000);
+        assertEquals(1, ParallelUtil.threadSize(batchSize, 0));
+    }
+
+    @Test
+    public void threadSizeShouldFailForZeroBatchsize() throws Exception {
+        try {
+            ParallelUtil.threadSize(0, randomInt());
```
Comment (jexp) [inline tests/src/test/java/org/neo4j/graphalgo/core/utils/ParallelUtilTest.java]: perhaps one for size 0 too. And some off by one. I.e. 999 and 1001
